protobuf/model_config.proto

```diff
@@ -814,6 +814,22 @@ message ModelOptimizationPolicy
     //@@       Currently only recognized by TensorRT backend.
     //@@
     bool output_copy_stream = 4;
+
```

Comment:
Can you move these changes (and below as well) to be parameters similar to the existing ones in the ort backend. Putting backend-specific parameters in this file is something we are trying to get away from.

Reply:
Do you mean these parameters in ORT backend?
```
optimization { execution_accelerators {
  gpu_execution_accelerator : [ {
    name : "tensorrt"
    parameters { key: "precision_mode" value: "FP16" }
    parameters { key: "max_workspace_size_bytes" value: "1073741824" }}]
}}
```

Well these are part of model_config.proto. I explicitly added a generic "map<string, string> parameters" so that these new fields are not runtime specific. Any runtime can use this field to add extra optimization configs.

Reply:
I added "Currently only recognized by ONNXRuntime backend." because the example parameters in comments are only recognized by onnxruntime. Perhaps I can rewords the comments.

Reply:
I realize that we have existing parameters that are marked as TensorRT only... and perhaps some of them could maybe be generalized for use in another backend... but that doesn't actually seem likely. So going forward we want backend parameters to use the existing generic parameters support, like here: https://github.com/triton-inference-server/tensorflow_backend#parameters

Reply:
Unless it is something we know that is very general, like the tensorrt optimization parameters

Reply:
Ok got it - You are referring to : https://github.com/triton-inference-server/common/blob/main/protobuf/model_config.proto#L1772 right?

Since these are all optimizations, I would like to add a generic parameters map to ModelOptimizationPolicy message. I can move the detailed comments out so that each backend can add the parameters they support in their repo documentation.